File: big-o-notation/exercises/reverse_string/reverse_string.py
def reverse_string(s: str) -> str:
    """
    Reverses a string using Python's slicing.
    Time Complexity: O(n)
    Space Complexity: O(n)
    """
    return s[::-1]

if __name__ == "__main__":
    test_str = "hello world"
    print(f"Original: {test_str}")
    print(f"Reversed: {reverse_string(test_str)}")

File: big-o-notation/exercises/reverse_string/reverse_string.ts
// reverse_string.ts

/**
 * Reverses a string using built-in array methods.
 * Time Complexity: O(n)
 * Space Complexity: O(n)
 */
export function reverseString(s: string): string {
    return s.split("").reverse().join("");
}

// Testing
const input = "hello world";
console.log(`Original: ${input}`);
console.log(`Reversed: ${reverseString(input)}`);
